rosbag2_tests/test/rosbag2_tests/process_execution_helpers.hpp

```diff
@@ -0,0 +1,152 @@
+// Copyright 2018, Bosch Software Innovations GmbH.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_TESTS__PROCESS_EXECUTION_HELPERS_HPP_
+#define ROSBAG2_TESTS__PROCESS_EXECUTION_HELPERS_HPP_
+
+#include <gmock/gmock.h>
+
+#include <signal.h>
+#include <stdlib.h>
+
+#include <chrono>
+#include <cstdlib>
+#include <string>
+
+using namespace ::testing;  // NOLINT
+
+#ifdef _WIN32
```

Comment:
with all that `#ifdef`, does it make sense to include two different helper files (one for win, one for unix) and only separate them once at the beginning?
```
#ifdef _WIN32
#include process_execution_helpers_windows.hpp
#else 
#include process_execution_helpers_unix.hpp
#endif 
```

Reply:
The current implementation has one big advantage:
It hides the fact that processes behave differently on windows and unix.

The includer therefore does not have to consider this and simply includes the one header available.

Reply:
Sorry, what I meant was having three files:

* `process_execution_helpers.hpp`
* `process_execution_helpers_windows.hpp`
* `process_execution_helpers_unix.hpp`

The first one having the code snippet above included.

Reply:
Ah, yes, that's a good idea, I added it.